oximeter/db/src/oxql/point.rs

```diff
@@ -1548,14 +1601,30 @@ impl<T: DistributionSupport> Distribution<T> {
             self.bins == rhs.bins,
             "Cannot subtract distributions with different bins",
         );
-        let counts = self
+        let counts: Vec<_> = self
             .counts
             .iter()
-            .zip(rhs.counts.iter().copied())
-            .map(|(x, y)| x.checked_sub(y))
+            .zip(rhs.counts.iter())
+            .map(|(x, y)| x.checked_sub(*y))
             .collect::<Option<_>>()
             .context("Underflow subtracting distributions values")?;
-        Ok(Self { bins: self.bins.clone(), counts })
+
+        // Subtract sum_of_samples and squared_mean directly.
+        // We allow underflow here.
+        let sum_of_samples = self.sum_of_samples - rhs.sum_of_samples;
```

Comment:
Yep. Will checked sub this, even though it will either be int64 or f64.

Reply:
Oh, right, these are signed, but the checked sub above is because the counts can't be negative. I was wrong then, this can definitely be negative. Suppose we get a huge negative sample in a later histogram? We want to reflect that in this subtraction.

Reply:
Ok, yeah, will leave this.